src/kit.ts

```diff
@@ -1172,6 +1172,8 @@ export async function effectiveKitEnvironment(kit: Kit, opts?: expand.ExpansionO
         }
     }
     if (kit.environmentSetupScript) {
+        /* TODO: Question, do we need override kit.environmentVariables with kit.environmentSetupScript or
+         override kit.environmentSetupScript with kit.environmentVariables, who has the higher priority */
```

Comment:
Good question.  I think it could go either way, but my vote would be for the `kit.environmentVariables` to have higher priority over `kit.environmentSetupScript` since I think of the script as being what runs first, then you can override it with static choices in the `environmentVariables` section of the Kit.